Components/Sensors/FastLogManager.cpp

```diff
@@ -0,0 +1,224 @@
+            // If the tick is past the slow period
+            if ((++msTick * AUTO_PT_SEND_PERIOD) > AUTO_PT_SLOW_PERIOD) {
+                // Sample and send
+                SamplePressureTransducer();
+                TransmitProtocolPressureData();
+                msTick = 0;
+            }
+
+            // Manage sending the log data
+            if(sendIdx < LOG_BUFFER_SIZE) {
+                PressureLog data = pressureLogBuffer[sendIdx];
+                TransmitLogData(data, sendIdx);
+                sendIdx += 1;
+            } else {
+                sendIdx = 0;
+            }
+            
+            // Delay
+            vTaskDelayUntil(&xLastWakeTime, AUTO_PT_SEND_PERIOD);
+            break;
+    }
+}
+
+
+/* Pressure Transducer Readers ------------------------------------------------------------------*/
+
+void ADC_Select_CH3 (void)
+{
+	ADC_ChannelConfTypeDef sConfig = {0};
+	  /** Configure for the selected ADC regular channel its corresponding rank in the sequencer and its sample time.
+	  */
+	  sConfig.Channel = ADC_CHANNEL_3;
+	  sConfig.Rank = 1;
+	  sConfig.SamplingTime = ADC_SAMPLETIME_28CYCLES;
+	  if (HAL_ADC_ConfigChannel(&hadc1, &sConfig) != HAL_OK)
+	  {
+	    Error_Handler();
+	  }
+}
+
+void ADC_Select_CH15 (void)
+{
+	ADC_ChannelConfTypeDef sConfig = {0};
+	  /** Configure for the selected ADC regular channel its corresponding rank in the sequencer and its sample time.
+	  */
+	  sConfig.Channel = ADC_CHANNEL_15;
+	  sConfig.Rank = 1;
+	  sConfig.SamplingTime = ADC_SAMPLETIME_28CYCLES;
+	  if (HAL_ADC_ConfigChannel(&hadc1, &sConfig) != HAL_OK)
+	  {
+	    Error_Handler();
+	  }
+}
+
+
+/**
+ * @brief This function reads and updates pressure readings
+ *          from the pressure transducer.
+ */
+void FastLogManager::SamplePressureTransducer()
+{
+	static const int PT_VOLTAGE_ADC_POLL_TIMEOUT = 50;
+	static const double PRESSURE_SCALE = 1.5220883534136546; // Value to scale to original voltage value
+	double adcVal[2] = {};
+	double pressureTransducerValue1 = 0;
+	double pressureTransducerValue2 = 0;
+	double vi = 0;
+
+	/* Functions -----------------------------------------------------------------*/
+	ADC_Select_CH3();
+	HAL_ADC_Start(&hadc1);  // Enables ADC and starts conversion of regular channels
+	if(HAL_ADC_PollForConversion(&hadc1, PT_VOLTAGE_ADC_POLL_TIMEOUT) == HAL_OK) { //Check if conversion is completed
+		adcVal[0] = HAL_ADC_GetValue(&hadc1); // Get ADC Value
+		HAL_ADC_Stop(&hadc1);
+		}
+	ADC_Select_CH15();
+		HAL_ADC_Start(&hadc1);  // Enables ADC and starts conversion of regular channels
+		if(HAL_ADC_PollForConversion(&hadc1, PT_VOLTAGE_ADC_POLL_TIMEOUT) == HAL_OK) { //Check if conversion is completed
+			adcVal[1] = HAL_ADC_GetValue(&hadc1); // Get ADC Value
+			HAL_ADC_Stop(&hadc1);
+			}
+	vi = ((3.3/4095) * (adcVal[0])); // Converts 12 bit ADC value into voltage
+	pressureTransducerValue1 = (250 * (vi * PRESSURE_SCALE) - 125) * 1000; // Multiply by 1000 to keep decimal places
```

Comment:
!! Fix conversion factor for the type. Eg) 10x PSI instead of 1000x PSI if keepingq 16-bit